.github/CONTRIBUTING.md

```diff
@@ -69,102 +67,142 @@ There are few ways you can play with the build:
 -  Playing with `webchat-loader`
    -  Navigate to https://compulim.github.io/webchat-loader/
    -  In the version dropdown, select `http://localhost:5000/webchat-es5.js`
--  Write your own web app
+-  Write your own HTML page to load Web Chat
    -  `<script src="http://localhost:5000/webchat.js"></script>`
-   -  Or using tarballs
+-  Create a new React app and symlink or load tarballs from these packages, in this specific order
+   1. `/packages/core`
+   1. `/packages/api`
+   1. `/packages/component`
+   1. `/packages/directlinespeech`
+   1. `/packages/bundle`
+
+# How to contribute to our code
 
-# Readying for publication
+1. [Write test pages first](#test-driven-development)
+1. Fix the issue or implement the new feature
+1. [Run static code analysis](#static-code-analysis)
+1. [Final checks](#final-checks)
+
+## Test driven development
 
 We care about software quality. Quality checking prevents regressions, reduces maintenance costs, minimizes chores, and enables us to move faster.
 
-Before submitting the pull request, please check your build by running these checks:
+For bugs, write test page to reproduce the bug, then fix it. This will prevent future regression.
 
--  [Running automated tests](#running-automated-tests)
--  [Performing static code analysis](#static-code-analysis)
+For features, use test page as a playground. Write new tests to verify different sub-features, e.g. rendering in carousel layout vs. rendering in stacked layout. Also write tests for both happy and unhappy paths. This will future-proof the work and protect the investment.
 
-## Running automated tests
+### Writing first test
 
-Automated tests in Web Chat are designed to run inside a stable and isolated test environment. To run the test suite, you will need to [start the test environment](#starting-the-test-environment), then [run the test suite](#running-the-test-suite).
+Start by copying the test page template from [`/__tests__/html/simple.html`](https://github.com/microsoft/BotFramework-WebChat/blob/main/__tests__/html/simple.html) and [`simple.js`](https://github.com/microsoft/BotFramework-WebChat/blob/main/__tests__/html/simple.js). Additionally, follow [other test pages](https://github.com/microsoft/BotFramework-WebChat/tree/main/__tests__/html) to learn about our [page object model](https://github.com/microsoft/BotFramework-WebChat/tree/main/packages/test/page-object/src/globals).
 
-### Starting the test environment
+We prefer using end-to-end visual regression tests (VRT) for pixel-perfect and whole feature verification. And unit tests for utility functions. For VRT, we use [`pixelmatch`](https://npmjs.com/package/pixelmatch) via [`jest-image-snapshot`](https://npmjs.com/package/jest-image-snapshot).
```

Comment:
```suggestion
We prefer using end-to-end visual regression tests (VRT) for pixel-perfect and whole feature verification. Unit tests should be written for utility functions. For VRT, we use [`pixelmatch`](https://npmjs.com/package/pixelmatch) via [`jest-image-snapshot`](https://npmjs.com/package/jest-image-snapshot).
```

Reply:
This is good. I want to learn how I can spot this mistake.

The original one looks too good to me and it blinded me. Any tips?

Reply:
That's hard. Generally, it's good to avoid starting a sentence with 'And' or 'But'. If you want to use 'But', I would instead use 'However' (beginning of the sentence). 

In beginner English, they say "never start a sentence with And or But". Then in intermediate, they say "It's actually ok to start with and or but, but don't use it too much".  I think there are more elegant solutions to convey the same. 

If you want to start a sentence with 'and', try removing the and and see how the flow is. Generally, if the sentence is a part of a paragraph, it is intrinsic that the two sentences are related.

(Unresolving for visibility. No action item)